sql/analyzer/triggers.go

```diff
@@ -401,13 +401,30 @@ func getTriggerLogic(ctx *sql.Context, a *Analyzer, n sql.Node, scope *plan.Scop
 		s := (*plan.Scope)(nil).NewScope(scopeNode).WithMemos(scope.Memo(n).MemoNodes()).WithProcedureCache(scope.ProcedureCache())
 		triggerLogic, _, err = a.analyzeWithSelector(ctx, trigger.Body, s, SelectAllBatches, noRowUpdateAccumulators)
 	case sqlparser.UpdateStr:
-		scopeNode := plan.NewProject(
-			[]sql.Expression{expression.NewStar()},
-			plan.NewCrossJoin(
-				plan.NewTableAlias("old", getResolvedTable(n)),
-				plan.NewTableAlias("new", getResolvedTable(n)),
-			),
-		)
+		// The scopeNode for an UpdateJoin should contain every node in the updateSource as new and old.
+		var scopeNode *plan.Project
+		if updateNode, isUpdate := n.(*plan.Update); isUpdate {
```

Comment:
one additional way to maybe make this clearer would be to do an if/else if we find updateJoin, which seems like the intended behavior? If there is an update join we want want the schema of the whole relation, otherwise we expect a single table source. Like I'm assuming we should error if we find an update join UpdateSource isn't the child? I wonder why it's not just hard-typed to UpdateSource